Funded an order (crypto-listing) and it caused the vendors node to panic and crash

This is the contract:

```
{
  "vendorListings": [
    {
      "slug": "ltc-bat",
      "vendorID": {
        "peerID": "Qma14uKNrz9rVZoKmXDq8439wPZg9o8wz968EucUfU1Lh4",
        "handle": "",
        "pubkeys": {
          "identity": "CAESIPRXZFIbHlWvcSDo592lVr/HT4z0WILdrW8NjWK6PJYQ",
          "bitcoin": "AxZsdkL0aBpo9CACnZYYSnzE5x6duy6RjwlIpyEy88Uh"
        },
        "bitcoinSig": "MEUCIQDtJluiQJA+cjmHgjlzgm1QKwmjVGFBWKoeE2e81uh0PQIgTZWEm+qmiJ5+f8NGeKOp5t+/waumdXrpdWqOei4JONk="
      },
      "metadata": {
        "version": 5,
        "contractType": "CRYPTOCURRENCY",
        "format": "MARKET_PRICE",
        "expiry": "2037-12-31T06:00:00.000Z",
        "acceptedCurrencies": [
          "LTC"
        ],
        "pricingCurrency": "",
        "language": "",
        "escrowTimeoutHours": 1080,
        "coinType": "BAT",
        "coinDivisibility": 8,
        "priceModifier": 0
      },
      "item": {
        "title": "LTC-BAT",
        "description": "TEST LISTING - DO NOT BUY!",
        "processingTime": "",
        "price": 0,
        "nsfw": false,
        "tags": [],
        "images": [
          {
            "filename": "Screen Shot 2016-02-14 at 5.12.03 PM.png",
            "original": "QmPkHKo32z2124FZDqEyh3Yv8Pu4MH9zup427gMkJGjAFb",
            "large": "QmTCpxCAwCSfrdFTMtTm4gj7sJbbXRTYJ3Psj44rLWWM84",
            "medium": "QmSUYL1PtDoc76TWaJ4xqGgJfUJqhoGxyXY1nSQpcuvkzj",
            "small": "QmW5ySc2juUMWwUyfQzwqLKixWrDmE55ELQDmY7r7D9ByR",
            "tiny": "QmYKBqbCf5uzYs2eBvu3XWfB5BR4yRxbssfXXDv3jgMfez"
          }
        ],
        "categories": [],
        "grams": 0,
        "condition": "",
        "options": [],
        "skus": [
          {
            "productID": "",
            "surcharge": 0,
            "quantity": 0,
            "bigSurcharge": "",
            "bigQuantity": "0"
          }
        ],
        "priceModifier": 1.25,
        "bigPrice": ""
      },
      "shippingOptions": [],
      "coupons": [],
      "moderators": [],
      "termsAndConditions": "",
      "refundPolicy": ""
    }
  ],
  "buyerOrder": {
    "refundAddress": "LS36pwru6DZk3PAtQYrm2dQpyf46eCJ8or",
    "refundFee": 0,
    "shipping": {
      "shipTo": "",
      "address": "",
      "city": "",
      "state": "",
      "postalCode": "",
      "country": "NA",
      "addressNotes": ""
    },
    "buyerID": {
      "peerID": "QmUhBVZDCcY9RzhnYXbQxUFETCG8ECvHcRfSGHiR5pecSJ",
      "handle": "",
      "pubkeys": {
        "identity": "CAESIPZOYuamDdHnp3c81KSAuvOTA08utkMUdNH+NtpmO+PW",
        "bitcoin": "AseU1TvTO42NBY+Hl4i5tjlgEP0WVTV4mCwPeKsYDKIY"
      },
      "bitcoinSig": "MEUCIQC4PsVy/SMXAA99e0MLOlX/W3LW0/zK2ui4gsftpGdiuQIgYn0ChDSf8GE1dOFnR0lTmI+dCbbLasViME0seJSKYOA="
    },
    "timestamp": "2019-10-29T19:40:00.668883Z",
    "items": [
      {
        "listingHash": "zb2rhoPimDAgSKgbVmYYQZqAjnsECTqHpEfk7DotSF7NW1anA",
        "quantity": 0,
        "memo": "",
        "paymentAddress": "1.2345 season at shennandoh",
        "quantity64": 0,
        "bigQuantity": "123450000"
      }
    ],
    "payment": {
      "method": "DIRECT",
      "moderator": "",
      "amount": 0,
      "chaincode": "a7df58e46464b4ae23ee60c8b4f2bf2e1a2e84c008e8f929eb4f23e44561e25e",
      "address": "ltc1quung2k4axae7pdlrqyqdkhjak6vwmjjkj9pegd386dagylzyafyqzecd4d",
      "redeemScript": "5121021b1cd2fd42d78e8a714718f49b4fa8a78c3c219e99cb182d6e93552aad9d9e9a2102a0ebd19d7be818d2e8040546a7ee79c831d4fc888bfefdd390b5ae264e9ec98252ae",
      "coin": "",
      "bigAmount": "497979",
      "amountCurrency": {
        "code": "LTC",
        "divisibility": 8
      }
    },
    "ratingKeys": [
      "A1i/SwzaNzT9EV+l1haCWmJIygyiS3MBxFiPiP81pUVn"
    ],
    "alternateContactInfo": "",
    "version": 2,
    "bigRefundFee": ""
  },
  "signatures": [
    {
      "section": "LISTING",
      "signatureBytes": "pg4O7TBbI2fmwGWHdmtlzNUKoOekKfnk+/RACPM6AoNxMP6en6qwH0aMx8kD0yrX162XEyZLKLjlCkGD8URcAw=="
    },
    {
      "section": "ORDER",
      "signatureBytes": "XDlKp5tEDHuWlh0kK9R115zmv7MyMp3C0r9GJerPr28SrKbXnEQM1g+xJHh/aJXJCw9LwjljrNDjMNp87q0lBg=="
    }
  ]
}
```

This is the panic:

```
14:02:26.580 [INFO] [transaction-listener/OnTransactionReceived] Transaction received 52e15b20f134c241d38f08dfc63f7b18de544fdc96e69f99e4bfd52d150b4d26 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5031dd2]

goroutine 51233 [running]:
github.com/OpenBazaar/openbazaar-go/wallet/listeners.currencyDivisibilityFromContract(0xc00129a060, 0xc00123ea80, 0xc0038ee8c8)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/wallet/listeners/transaction_listener.go:313 +0x42
github.com/OpenBazaar/openbazaar-go/wallet/listeners.(*TransactionListener).processSalePayment(0xc0005a9e90, 0xc002722440, 0x40, 0x5a30120, 0xc002e7e940, 0x0, 0xc001a3f7d0, 0x1, 0x5, 0x0, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/wallet/listeners/transaction_listener.go:229 +0xab3
github.com/OpenBazaar/openbazaar-go/wallet/listeners.(*TransactionListener).OnTransactionReceived(0xc0005a9e90, 0xc002722440, 0x40, 0xc003844ab0, 0x2, 0x2, 0xc002568180, 0x4, 0x4, 0x0, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/wallet/listeners/transaction_listener.go:78 +0x1cae
github.com/OpenBazaar/openbazaar-go/wallet/listeners.(*TransactionListener).OnTransactionReceived-fm(0xc002722440, 0x40, 0xc003844ab0, 0x2, 0x2, 0xc002568180, 0x4, 0x4, 0x0, 0xbf664494a273ea70, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/cmd/start.go:679 +0x58
github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service.(*WalletService).callbackListeners(0xc00129d9e0, 0xc002722440, 0x40, 0xc003844ab0, 0x2, 0x2, 0xc002568180, 0x4, 0x4, 0x0, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go:561 +0x85
github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service.(*WalletService).saveSingleTxToDB(0xc00129d9e0, 0xc003d56ec0, 0x40, 0x1, 0x0, 0xc002347200, 0x4, 0x4, 0xc0029bad80, 0x2, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go:545 +0xdcf
github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service.(*WalletService).ProcessIncomingTransaction(0xc00129d9e0, 0xc003d56ec0, 0x40, 0x1, 0x0, 0xc002347200, 0x4, 0x4, 0xc0029bad80, 0x2, ...)
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go:151 +0x216
created by github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service.(*WalletService).listen
	/Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go:137 +0x377
exit status 2
```

--- a/wallet/listeners/transaction_listener.go
+++ b/wallet/listeners/transaction_listener.go
@@ -226,12 +226,14 @@ func (l *TransactionListener) processSalePayment(txid string, output wallet.Tran
 		}
 
 		// update divisibility from contract listing
-		customDivisibility := currencyDivisibilityFromContract(l.multiwallet, contract)
-		if customDivisibility != currencyValue.Currency.Divisibility {
-			currencyValue.Currency.Divisibility = customDivisibility
-			if err := currencyValue.Valid(); err != nil {
-				log.Errorf("Invalid currency divisibility (%d) found in contract (%s): %s", customDivisibility, orderId, err.Error())
-				return
+		if contract.VendorListings[0].Metadata.ContractType != pb.Listing_Metadata_CRYPTOCURRENCY {
+			customDivisibility := currencyDivisibilityFromContract(l.multiwallet, contract)
+			if customDivisibility != currencyValue.Currency.Divisibility {
+				currencyValue.Currency.Divisibility = customDivisibility
+				if err := currencyValue.Valid(); err != nil {
+					log.Errorf("Invalid currency divisibility (%d) found in contract (%s): %s", customDivisibility, orderId, err.Error())
+					return
+				}
 			}
 		}
 

